recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,112 @@
-        return []
+    def auto_tools(self):
+        if self._auto_tools is not None:
+            return self._auto_tools
+        self._auto_tools = Autotools(self)
+        return self._auto_tools
 
     def build(self):
         # If building on x86_64, change the default directory name for 64-bit libraries to "lib":
-        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
-        tools.replace_in_file(os.path.join(self.source_folder,
-                                           self._source_subfolder, "gcc", "config", "i386", "t-linux64"),
-                              "m64=../lib64", "m64=../lib", strict=False)
+        replace_in_file(self, os.path.join(self.source_folder, "gcc", "config", "i386", "t-linux64"),
+                        "m64=../lib64", "m64=../lib", strict=False)
         # Ensure correct install names when linking against libgcc_s;
         # see discussion in https://github.com/Homebrew/legacy-homebrew/pull/34303
-        tools.replace_in_file(os.path.join(self.source_folder,
-                                           self._source_subfolder, "libgcc", "config", "t-slibgcc-darwin"),
-                              "@shlib_slibdir@", libdir, strict=False)
-        autotools = self._configure_autotools()
-        autotools.make(args=self._make_args)
+        replace_in_file(self,
+                        os.path.join(self.source_folder, 'libgcc', 'config', 't-slibgcc-darwin'),
+                        '@shlib_slibdir@',
+                        os.path.join(self.package_folder, 'lib'),
+                        strict=False)
+        self.auto_tools.configure()
+        self.auto_tools.make()
 
     def package_id(self):
         del self.info.settings.compiler
 
     def package(self):
-        autotools = self._configure_autotools()
-        if self.settings.build_type == "Debug":
-            autotools.install(args=self._make_args)
+        if self.settings.build_type == 'Debug':
+            self.auto_tools.install()
         else:
-            autotools.make(args=["install-strip"] + self._make_args)
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.remove_files_by_mask(self.package_folder, "*.la")
-        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+            self.auto_tools.make('install-strip')
+        rmdir(self, os.path.join(self.package_folder, 'share'))
+        rm(self, '*.la', self.package_folder, recursive=True)
+        copy(self, pattern='COPYING*', dst=os.path.join(self.package_folder, 'licenses'), src=self.source_folder)
 
     def package_info(self):
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : " + bindir)
-        self.env_info.PATH.append(bindir)
+        bindir = os.path.join(self.package_folder, 'bin')
+        self.output.info('Appending PATH env var with : ' + bindir)
+        self.env_info.PATH = [bindir] + self.env_info.PATH
```

Comment:
This is unsual... what exactly is it for?

Reply:
made this more Conan v2 style. I think bin dir needs to be on the path if you actually want to build anything with gcc.

Reply:
Where did you find the "2.0 style"?

As a c++ dev this is way less readable. I had to Google what this does 🙊 the old append was more obvious but maybe I need to work on my python skill